controllers/service_controller.go

```diff
@@ -110,6 +117,69 @@ func (r *ServiceReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
 	return ctrl.Result{}, nil
 }
 
+func (r *ServiceReconciler) isOASDefined(ctx context.Context, service *corev1.Service, log logr.Logger) (bool, string, error) {
+	var oasContent string
+	if oasConfigmapName, ok := service.Annotations[KuadrantDiscoveryAnnotationOASConfigMap]; ok {
+		oasContent, err := r.fetchOpenAPIFromConfigMap(
+			ctx,
+			types.NamespacedName{Name: oasConfigmapName, Namespace: service.Namespace},
+		)
+		log.V(1).Info("get OAS configmap", "objectKey", types.NamespacedName{Name: oasConfigmapName, Namespace: service.Namespace}, "error", err)
+		return true, oasContent, err
+	}
+
+	// The annotations for authodiscovery are the following, if not namedPort, it'll use the first one.
+	// discovery.kuadrant.io/oas-path: "/openapi"
+	// discovery.kuadrant.io/oas-port: openapi
+	if oasURLdefined, ok := service.Annotations[KuadrantDiscoveryAnnotationOASPath]; ok {
+		var targetPort int32 = 80
+		if oasNamePort, ok := service.Annotations[KuadrantDiscoveryAnnotationOASNamePort]; ok {
+			for _, port := range service.Spec.Ports {
+				if port.Name == oasNamePort {
+					targetPort = port.Port
+					// TODO: log messsage here
+					break
+				}
+			}
+		} else {
+			if len(service.Spec.Ports) > 0 {
+				targetPort = service.Spec.Ports[0].Port
+			}
+		}
+
+		// This is terrible hack, we should use the same way as Coredns use it, but
+		// for the moment is ok like this
+		// About coredns, you can see some examples here, externalName, etc..
+		// https://github.com/coredns/coredns/blob/4758070425ac2f4f43c71c82ba3aca847965e394/plugin/kubernetes/kubernetes.go#L527
+		var targetURL = fmt.Sprintf("http://%s.%s.svc.cluster.local:%d", service.Name, service.Namespace, targetPort)
+		resp, err := http.Get(fmt.Sprintf("%s%s", targetURL, oasURLdefined))
+		if err != nil {
+			return true, "", err
+		}
+		defer resp.Body.Close()
+
+		// We read the body here, just before checking the status code, just
+		// because the team want to log the body in case of a invalid status code.
+		// Context: https://github.com/Kuadrant/kuadrant-controller/pull/44#discussion_r684146743
+		body, err := io.ReadAll(resp.Body)
+		if err != nil {
+			return true, "", fmt.Errorf("cannot read the body of %s: %w", targetURL, err)
+		}
+
+		if resp.StatusCode != 200 {
+			var bodyLogMax = maxBodyLog
+			if cap(body) < bodyLogMax {
+				bodyLogMax = cap(body)
```

Comment:
use `len`. `cap` is the capacity that the array/slice has currently to host data. `len` and `cap` are not necessarily equal.

Reply:
The thing is, you don't want that bodyLogMax is bigger than capacity, so that it triggers a panic. This is why I use cap instead of len, I'll change